src/core/utils/CorePlugin.ts

```diff
@@ -39,19 +40,33 @@ export class CorePlugin extends JWPlugin {
         // TODO: check the intent to insert other things than text.
         this.editor.vDocument.insertText(intent.payload.value);
     }
+    /**
+     * Remove in the backward direction (backspace key expected behavior).
+     */
     removeBackward(): void {
-        // TODO: this is a stub
         const range = this.editor.vDocument.range;
         if (range.isCollapsed()) {
-            range.setStart(range.start.previous());
+            const previous = VDocument.withRange(() => range.start.previousLeaf());
```

Comment:
Why would you want that `previousLeaf` may return a range node ? (same for `removeForward`)

Reply:
Handled in https://github.com/odoo-dev/jabberwock/pull/33/commits/27895a6e7515bc45f4920c99d68660e1ec104643#diff-a011953d0e7ac92c7b6d27420365d566R48